src/DB/Migration/MigrationVersion141.php

```diff
@@ -0,0 +1,69 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\DB\Migration;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\DB\DBHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\DB\Table\MerchantIssueTable;
+use wpdb;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class MigrationVersion1_4_1
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\DB\Migration
+ *
+ * @since x.x.x
+ */
+class MigrationVersion141 extends AbstractMigration {
+
+	/**
+	 * @var DBHelper
+	 */
+	protected $db_helper;
+
+	/**
+	 * @var MerchantIssueTable
+	 */
+	protected $mc_issues_table;
+
+	/**
+	 * MigrationVersion141 constructor.
+	 *
+	 * @param wpdb               $wpdb The wpdb object.
+	 * @param DBHelper           $db_helper
+	 * @param MerchantIssueTable $mc_issues_table
+	 */
+	public function __construct( wpdb $wpdb, DBHelper $db_helper, MerchantIssueTable $mc_issues_table ) {
+		parent::__construct( $wpdb );
+		$this->db_helper       = $db_helper;
+		$this->mc_issues_table = $mc_issues_table;
+	}
+
+
+	/**
+	 * Returns the version to apply this migration for.
+	 *
+	 * @return string A version number. For example: 1.4.1
+	 */
+	public function get_applicable_version(): string {
+		return '1.4.1';
+	}
+
+	/**
+	 * Apply the migrations.
+	 *
+	 * @return void
+	 */
+	public function apply(): void {
+		if ( $this->db_helper->table_exists( $this->mc_issues_table->get_name() ) &&
+			 $this->db_helper->index_exists( $this->mc_issues_table->get_name(), 'product_issue' )
```

Comment:
I originally didn't add it there because I thought it would make it bloated. I mean if we add `index_exists` then the same could apply for `add_index` or `drop_index` or many other DB methods that we might need.

Even though there are a couple of DB methods already in there, I think adding more methods will broaden its scope and make it handle all DB table operations. I think I rather have `TableInterface` just represent a table in our system and not handle DB operations because that's a whole other set of functionality.

That said, I've added the `TableInterface::has_index` and removed the DBHelper class in https://github.com/woocommerce/google-listings-and-ads/pull/958/commits/ebd6286c47454614336dceed151fc83f19dbbc7d. Let me know what you think.

Reply:
Thanks for the explanation. That makes sense, it just seemed overkill to have a helper just for one function. `add_index` is supported by dbDelta so we shouldn't need it. `drop_index` in this PR is just left to the 1.4.1 migration class, I'm not sure how often we might need it in the future, but hopefully not often enough to need a helper function for it. By the same logic we could confine the `index_exists` to a protected function within the migration class. But I personally don't think it's bloating the TableInterface too much.